scripts/cnode-helper-scripts/mithril-signer.sh

```diff
@@ -97,6 +97,8 @@ deploy_systemd() {
 	StartLimitIntervalSec=0
 	Wants=network-online.target
 	After=network-online.target
+	BindsTo=${vname}.service
```

Comment:
From what I understand it does not require binding to cnode , and does not have a strong dependency on local client (if cannot connect on startup, it would fail independently). Do you foresee a special case where this will be needed? Just thinking ahead, if not necessary - it's best to avoid adding a dependency.

PS: I know a small subset of other systemd services does have `BindsTo` and `After` , but fwir they have cases which lead to unhandled loop, it shouldnt be the case for this service [similar to submit-api and ogmios].

Reply:
Mithril signer relies on the Cardano node socket if it starts before the node the service will result in the following failure.
```
Error: "Could not get Epoch: general error Error launching command Command { std: CARDANO_NODE_SOCKET_PATH=\"/opt/cardano/cnode/sockets/node0.socket\" \"/app/bin/cardano-cli\" \"query\" \"tip\" \"--testnet-magic\" \"1\", kill_on_drop: false }, error = '\ncardano-cli: Network.Socket.connect: <socket: 11>: does not exist (No such file or directory)'"
```

My using containers, and not the Systemd cnode.service unit, I do not recall if the service shows Active only after the socket is up or beore. Mithril signer having `After` is to eliminate, or at least reduce, the errors from Mithril signer in the system logs. 

I went with `BindsTo` as I did not observe the use of `Requires`, so figured it was a style or standardization choice. Also in every instance where a Systemd service unit in `deploy-as-systemd.sh` had an `After=cnode.service` statement it also included a `BindsTo=cnode.service` statement. Given this would ensure startup after cnode when the service unit is deployed, and if the node was stopped would also stop the Mithril signer so it wasn't spamming the logs every `RestartSec` until the node was back online, this felt like a good combination for a production Mithril signer + cnode Sytemd setup.

Reply:
I think bindsTo in this case makes sense as explained so from my end 👍